Code/Framework/AzFramework/AzFramework/Process/ProcessWatcher.cpp

```diff
@@ -15,14 +15,63 @@
 
 namespace AzFramework
 {
+    namespace ProcessLauncher
+    {
+        AZStd::string ProcessLaunchInfo::GetCommandLineParametersAsString() const
+        {
+            struct CommandLineParametersVisitor
+            {
+                AZStd::string operator()(const AZStd::string& commandLine) const
+                {
+                    return commandLine;
+                }
+
+                AZStd::string operator()(const AZStd::vector<AZStd::string>& commandLineArray) const
+                {
+                    AZStd::string commandLineResult;
+
+#if AZ_TRAIT_OS_USE_WINDOWS_FILE_PATHS
+                    // When re-constructing a command line from an argument list (on windows), if an argument
+                    // is double-quoted, then the double-quotes must be escaped properly otherwise
+                    // it will be absorbed by the native argument parser and possibly evaluated as
+                    // multiple values for arguments
+                    AZStd::string_view escapedDoubleQuote = R"("\")";
+
+                    AZStd::vector<AZStd::string> preprocessed_command_array;
+                    for (const auto& commandArg : commandLineArray)
+                    {
+                        AZStd::string replacedArg = commandArg;
+                        auto replacePos = replacedArg.find_first_of('"');
+                        while (replacePos != replacedArg.npos)
+                        {
+                            replacedArg.replace(replacePos, 1, escapedDoubleQuote.data(), escapedDoubleQuote.size());
+                            replacePos += escapedDoubleQuote.size();
+                            replacePos = replacedArg.find('"', replacePos);
+                        }
+                        preprocessed_command_array.emplace_back(replacedArg);
+                    }
+                    AzFramework::StringFunc::Join(
+                        commandLineResult, preprocessed_command_array.begin(), preprocessed_command_array.end(), " ");
+#else
+                    AzFramework::StringFunc::Join(commandLineResult, commandLineArray.begin(), commandLineArray.end(), " ");
+#endif // AZ_TRAIT_AZFRAMEWORK_DOUBLE_ESCAPE_COMMANDLINE_QUOTES
+
+                    return commandLineResult;
+                }
+            };
+
+            return AZStd::visit(CommandLineParametersVisitor{}, m_commandlineParameters);
+        }
+    }
```

Comment:
Can you PALify this function into ProcessWatcher_<platform>.cpp?

Reply:
I started to but hesitated to add so many changes to do a windows specific thing